### DISCUSSION TITLE
Output at each fixed point iteration

### DISCUSSION THREAD

**Discussion by BoZeng1997** [idaholab/moose#21117] [Q&A General] [answered]
Hi,

I want to know if there is a way to output exodus file at each picard iteration within a timestep? Or where can I find an example of creating custom flag for picard iteration?

Bo

**Comment by GiudGiud** [comment, marked as answer]
Hello

we currently do not have this feature. 
if this is just for debugging I d recommend setting the max number of fixed point iterations at all the desired times (running the code multiple times) then setting the accept_on_max_fixed_point_iteration parameter of the executioner to true

guillaume

**Reply by BoZeng1997** [reply]
Not for debugging. I want to see how picard iteration find its path. The cost to run it till the critical step is high and the iteration in that step is of hundreds. Is there a way I can probably create an object to achieve this?

**Reply by GiudGiud** [reply]
If you’re open to coding it s not that difficult to create an execute_on flag to output on fixed point iterations. 
@lindsayad has something like this in one of his branches I think

**Reply by BoZeng1997** [reply]
is Alex Lindsay open to inquire here? If so, shall I wait till he replies? I will be grateful if I can follow an example.

**Reply by GiudGiud** [reply]
He s busy right now I believe but he ll get back to us when he has time

**Reply by lindsayad** [reply]
This is what I have for an example: lindsayad/moose@7d0e9a3228cd221

**Reply by BoZeng1997** [reply]
Hi @lindsayad . Is there a way to pack each iteration within a step into one exodus file in sequence? I did not find the way to append iteration step in exodus output.

**Reply by lindsayad** [reply]
Probably not for Picard iterations. What would you even want to write the timestep as?
